File: benchmark-nodejs/routes/routing.js
// Router Middleware
const express = require("express");
const router = express.Router();
const path = require("path");

// Routing Test
router.get("/hello", function(req, res) {
  res.send("Hello World!");
});

router.get("/hello/json", function(req, res) {
  res.json({message: "Hello World!"});
});

router.get("/hello/html", function(req, res) {
  res.sendFile(path.join(__dirname + "/../templates/hello.html"));
});

router.get("/hello/html/:msg", function(req, res) {
  res.render("hello_msg", {msg: req.params["msg"]})
});

router.get("/hello/msg/:msg", function(req, res) {
  res.send("Hello World! - " + req.params["msg"]);
});

router.get("/hello/query", function(req, res) {
  res.send("Hello World! - " + req.query["msg"]);
});

router.get("/badrequest", function(req, res) {
  res.status(400).send("Successful response!");
});

router.get("/list", function(req, res) {
  let text = "";
  for (var x = 1; x <= 5000; x++) {
    text += "This is the line " + x + "\r\n<br>";
  }
  res.send(text);
});

module.exports = router;

File: benchmark-flask/BenchmarkFlask/routing.py
from flask import render_template, jsonify, request
from BenchmarkFlask import app

@app.route("/hello")
def hello():
  return "Hello World!"

@app.route("/hello/json")
def hello_json():
  data = {'message': 'Hello World!'}
  return jsonify(data)

@app.route("/hello/html")
def html_method():
  msg = ""
  return render_template("hello.html", msg = msg)

@app.route("/hello/html/<msg>")
def html_msg_method(msg):
  return render_template("hello_msg.html", msg = msg)

@app.route("/hello/msg/<msg>")
def msg(msg):
  return "Hello World! - " + msg

@app.route("/hello/query")
def query_method():
  msg = request.args.get("msg")
  return "Hello World! - " + msg

@app.route("/badrequest")
def badrequest_method():
  return "Successful response!", 400

@app.route("/list")
def list_method():
  text = ""
  for i in range(5001):
    text += "This is the line " + str(i) + "\n<br>"

  return text